bundles/org.eclipse.mita.base/model/expressions.xcore

```diff
@@ -129,90 +155,170 @@ class AssignmentExpression extends Expression {
 	contains Expression varRef
 	AssignmentOperator operator
 	contains Expression expression
+	op String toString() {
+		return varRef.toString + " = " + expression.toString
+	}
 }
 class ConditionalExpression extends Expression {
 	contains Expression condition
 	contains Expression trueCase
 	contains Expression falseCase
+	op String toString() {
+		return condition + " ? " + trueCase + " : " + falseCase;	
+	}
 }
 class LogicalOrExpression extends BinaryExpression {
 	op EEnumerator getOperator() {
 		return LogicalOperator.OR
 	}
+	op String toString() {
+		return super.toString;
+	}
 }
 class LogicalAndExpression extends BinaryExpression { 
 	op EEnumerator getOperator() {
 		return LogicalOperator.AND
 	}
+	op String toString() {
+		return super.toString;
+	}
 }
 class LogicalNotExpression extends UnaryExpression {
 	op EEnumerator getOperator() {
 		return LogicalOperator.NOT
+	}
+	op String toString() {
+		return super.toString;
 	} 
 }
 class BitwiseXorExpression extends BinaryExpression {
 	op EEnumerator getOperator() {
 		return BitwiseOperator.XOR
 	}  
+	op String toString() {
+		return super.toString;
+	}
 }
 class BitwiseOrExpression extends BinaryExpression {
 	op EEnumerator getOperator() {
 		return BitwiseOperator.OR
 	} 
+	op String toString() {
+		return super.toString;
+	}
 }
 class BitwiseAndExpression extends BinaryExpression {
 	op EEnumerator getOperator() {
 		return BitwiseOperator.AND
 	} 
+	op String toString() {
+		return super.toString;
+	}
 }
 class LogicalRelationExpression extends BinaryExpression {
-	RelationalOperator operator 
+	RelationalOperator operator
+	op String toString() {
+		return super.toString;
+	} 
 }
 class ShiftExpression extends BinaryExpression {
 	ShiftOperator operator 
+	op String toString() {
+		return super.toString;
+	}
 }
 class NumericalAddSubtractExpression extends BinaryExpression {
 	AdditiveOperator operator 
+	op String toString() {
+		return super.toString;
+	}
 }
 class NumericalMultiplyDivideExpression extends BinaryExpression {
 	MultiplicativeOperator operator 
+	op String toString() {
+		return super.toString;
+	}
 }
 class NumericalUnaryExpression extends UnaryExpression {
 	UnaryOperator operator 
+	op String toString() {
+		return super.toString;
+	}
```

Comment:
Is there any need for all these toString methods that just call super?

Reply:
yes, because the default behaviour is to NOT call it for Ecore/Xcore objects.